core/templates/dev/head/pages/collection_editor/editor_tab/CollectionNodeCreatorDirective.js

```diff
@@ -25,17 +25,40 @@ oppia.directive('collectionNodeCreator', [function() {
       'validatorsService', 'CollectionEditorStateService',
       'CollectionLinearizerService', 'CollectionUpdateService',
       'CollectionNodeObjectFactory', 'ExplorationSummaryBackendApiService',
-      'siteAnalyticsService',
+      'siteAnalyticsService', 'oppiaDebouncer',
       function(
           $scope, $http, $window, $filter, alertsService,
           validatorsService, CollectionEditorStateService,
           CollectionLinearizerService, CollectionUpdateService,
           CollectionNodeObjectFactory, ExplorationSummaryBackendApiService,
-          siteAnalyticsService) {
+          siteAnalyticsService, oppiaDebouncer) {
         $scope.collection = CollectionEditorStateService.getCollection();
         $scope.newExplorationId = '';
         $scope.newExplorationTitle = '';
         var CREATE_NEW_EXPLORATION_URL_TEMPLATE = '/create/<exploration_id>';
+        $scope.searchQuery = '';
+
+        var onSearchQueryChangeExec = function  () {
+          queryUrl = '/exploration/metadata_search' + '?q=' + $scope.searchQuery;
+          $http.get(queryUrl).then(function(response) {
+            var data = response.data;
+            var options = '';
+            for(var i = 0; i < data.collection_node_metadata_list.length; i++)
+              if(data.collection_node_metadata_list[i]) {
+                options += '<option value=" ' + data.collection_node_metadata_list[i].id + '">' + data.collection_node_metadata_list[i].title + '</option>';
```

Comment:
Please do not do any HTML construction in JS. All HTML construction should happen in the DOM itself.